lib/macho/load_commands.rb

```diff
@@ -268,7 +269,9 @@ def initialize(lc, lc_str)
         if view
           lc_str_abs = view.offset + lc_str
           lc_end = view.offset + lc.cmdsize - 1
-          @string = view.raw_data.slice(lc_str_abs..lc_end).delete("\x00")
+          raw_string = view.raw_data.slice(lc_str_abs..lc_end)
+          @string, null_byte, _ = raw_string.partition("\x00")
+          raise LCStrMalformedError.new(lc) if @string.empty? || null_byte.empty?
```

Comment:
> I made the condition `@string.empty? || null_byte.empty?`, since we probably also want to prevent the user from doing something like `change_install_name(old, "\x00")`.

That's a good intention, but I think we have to tackle validation of user input separately (and more thoroughly, if we want to go down that route). Let's focus on the bug fix here!

That's also not the code path being taken if the user invokes `LoadCommand.create` (or any of the helper methods that make use of it). That case is handled in the `else` branch below. Here we're dealing with data coming directly from the binary.